backend/src/model/model.service.ts

```diff
@@ -0,0 +1,55 @@
+import { pipeline, env, PipelineType } from "@huggingface/transformers";
+import { ConfigService } from "../config/config.service";
+
+type Progress = { loaded: number; total: number };
+type ProgressCallback = (progress: Progress) => void;
+
+env.allowLocalModels = false;
+
+class PipelineSingleton {
+    static pipelines = new Map<string, any>();
+
+    static async getPipeline(task: string, model: string, progressCallback?: ProgressCallback) {
+        const pipelineOptions = progressCallback ? { progress_callback: progressCallback } : undefined;
+
+        const pipelineInstance = await pipeline(task as PipelineType, model, pipelineOptions);
+
+        if (!this.pipelines.has(model)) {
+            this.pipelines.set(model, pipelineInstance);
+        }
+        return this.pipelines.get(model);
+    }
+}
+
+export default async function loadAllChatsModels(progressCallback: ProgressCallback = () => {}) {
+    const configService = new ConfigService();
+    const chats = configService.get("chats");
```

Comment:
this should be exist in config loader, we should not prepare a unready config to other service